### PR TITLE
Cometbft: enable prometheus exporter

### PR DESCRIPTION
**Pull request by WafflesVonMaple**

### CODE REVIEW

**Comment by WafflesVonMaple** [comment]
@joroshiba worth chatting on this one since we are kinda crafting cometbft ourselves here. Worth looking at https://github.com/cometbft/cometbft/issues/899

File: charts/sequencer/values.yaml [hunk 1]
```diff
@@ -80,9 +80,17 @@ images:
...
   cometBFTRPC: 26657
+  cometBFTMETRICS: 26660
   sequencerABCI: 26658
   relayerRPC: 2450
 
+# ServiceMonitor configuration
+serviceMonitor:
+  enabled: true
+  port: 26660
+  labels:
```
Comment (WafflesVonMaple) [inline charts/sequencer/values.yaml:91]: need to change to additionalLabels per servicemonitor template

File: charts/sequencer/templates/_helpers.tpl [hunk 2]
```diff
@@ -30,3 +30,26 @@ Return the appropriate apiVersion for ingress.
...
+{{ include "sequencer.selectorLabels" . }}
+{{- end }}
+
+{{/*
+Selector labels
+*/}}
+{{- define "sequencer.selectorLabels" -}}
+app: {{ include "sequencer.name" . }}
+name: {{ .Values.config.moniker }}-sequencer-metrics
+{{- end }}
```
Comment (joroshiba) [inline charts/sequencer/templates/_helpers.tpl:55]: nit: new line

File: charts/sequencer/files/cometbft/config/config.toml [hunk 3]
```diff
@@ -456,7 +456,7 @@ psql-conn = ""
 # When true, Prometheus metrics are served under /metrics on
 # PrometheusListenAddr.
 # Check out the documentation for the list of available metrics.
-prometheus = false
+prometheus = true
```
Comment (joroshiba) [inline charts/sequencer/files/cometbft/config/config.toml:459]: Should we have this be the value of `.Values.serviceMonitor.enabled`?

Side note that we need to figure out how we can get the init to work on second deployment to update this file we copy (or get rid of the copy ideally)

File: charts/sequencer/templates/service.yaml [hunk 4]
```diff
@@ -34,3 +34,17 @@ spec:
...
+  name: {{ .Values.config.moniker }}-sequencer-metrics
+  namespace: {{ .Values.global.namespace }}
+spec:
+  selector:
+    app: {{ .Values.config.moniker }}-sequencer
+  ports:
+    - name: cometbft-metrics
+      port: {{ .Values.ports.cometBFTMETRICS }}
+      targetPort: cometbft-metrics
+{{- end }}
```
Comment (joroshiba) [inline charts/sequencer/templates/service.yaml:50]: nit: new line

File: charts/sequencer/templates/servicemonitor.yaml [hunk 5]
```diff
@@ -0,0 +1,28 @@
...
+  endpoints:
+    - port: {{ .Values.serviceMonitor.port }}
+      path: /metrics
+      {{- with .Values.serviceMonitor.interval }}
+      interval: {{ . }}
+      {{- end }}
+      {{- with .Values.serviceMonitor.scrapeTimeout  }}
+      scrapeTimeout: {{ . }}
+      {{- end }}
+{{- end }}
```
Comment (joroshiba) [inline charts/sequencer/templates/servicemonitor.yaml:28]: nit: new line

File: charts/sequencer/values.yaml [hunk 6]
```diff
@@ -80,9 +80,17 @@ images:
 ports:
   cometBFTP2P: 26656
   cometBFTRPC: 26657
+  cometBFTMETRICS: 26660
   sequencerABCI: 26658
   relayerRPC: 2450
 
+# ServiceMonitor configuration
+serviceMonitor:
+  enabled: true
```
Comment (joroshiba) [inline charts/sequencer/values.yaml:89]: Will having this enabled by default cause issues when coreos crds not installed?

Should we add some just commands to install prometheus, and probably by default disable?

File: charts/sequencer/templates/service.yaml [hunk 7]
```diff
@@ -34,3 +34,17 @@ spec:
...
 ---
+{{- if .Values.serviceMonitor.enabled }} 
+kind: Service
+apiVersion: v1
+metadata:
+  name: {{ .Values.config.moniker }}-sequencer-metrics
+  namespace: {{ .Values.global.namespace }}
+spec:
+  selector:
+    app: {{ .Values.config.moniker }}-sequencer
```
Comment (joroshiba) [inline charts/sequencer/templates/service.yaml:45]: Do we need to update this and other areas to use the new helpers name and labels?

**Review by joroshiba — CHANGES_REQUESTED** [review]
mostly new lines to be fixed, couple open questions.

File: charts/sequencer/values.yaml (hunk 6 above)
Comment (WafflesVonMaple) [reply charts/sequencer/values.yaml]: left in accidentally when testing local, set to false

File: charts/sequencer/templates/service.yaml (hunk 7 above)
Comment (WafflesVonMaple) [reply charts/sequencer/templates/service.yaml:45]: we dont have to do it this way at all either, was just a fast way to ensure service monitor selectors picked up the right app name

**Review by joroshiba — APPROVED** [review]

**Pull request merged**